File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/di/modules/MainModule.java
package com.woowrale.jmvp.di.modules;

import android.content.Context;
import com.woowrale.jmvp.MainApplication;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class MainModule {

    private final MainApplication mainApplication;

    public MainModule(MainApplication mainApplication) {
        this.mainApplication = mainApplication;
    }

    @Provides
    @Singleton
    Context provideContext(){
        return mainApplication;
    }

}

File: KCleanArchitectureApp/appdagger/src/main/java/com/woowrale/kcleanarchitecture/di/modules/MainModule.kt
package com.woowrale.kcleanarchitecture.di.modules

import android.content.Context
import com.woowrale.kcleanarchitecture.MainApplication
import com.woowrale.kcleanarchitecture.di.threads.JobThread
import com.woowrale.kcleanarchitecture.di.threads.UIThread
import com.woowrale.usecase.threads.JobScheduler
import com.woowrale.usecase.threads.UIScheduler
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class MainModule constructor(private var mainApplication: MainApplication) {

    @Provides
    @Singleton
    fun provideContext(): Context {
        return mainApplication
    }

    @Provides
    @Singleton
    fun provideJobScheduler(jobThread: JobThread): JobScheduler {
        return jobThread
    }

    @Provides
    @Singleton
    fun provideUIScheduler(uiThread: UIThread): UIScheduler {
        return uiThread
    }
}
